File: src/2733.neither-minimum-nor-maximum.2836/solution.ts
function findNonMinOrMax (nums: number[]): number {
    const min = Math.min(...nums);
    const max = Math.max(...nums);
    for (const n of nums) {
        if (n !== min && n !== max) {
            return n;
        }
    }
    return -1;
}

File: src/2733.neither-minimum-nor-maximum.2836/solution.java
class Solution {
    public int findNonMinOrMax(int[] nums) {
        int max = nums[0];
        int min = nums[0];
        for (int n : nums) {
            max = Math.max(max, n);
            min = Math.min(min, n);
        }
        for (int n : nums) {
            if (n != max && n != min) {
                return n;
            }
        }
        return -1;
    }
}